### DISCUSSION TITLE
Setting the current room temperature possible?

### DISCUSSION THREAD

**Discussion by DanMike** [dewenni/ESP_Buderus_KM271#44] [Ideas]
Is it possible to set/overwrite the current room temperature through KM271 or the board from the78mole?

Normally, the room temperature is read from the Buderus BFU room unit, which is connected by a proprietary 2-wire bus to the Logamatic 2107. Now I would like to set the current room temperature to (a) value(s) read from temperature sensors in other rooms than the one, where the BFU is installed.
I'm currently using FHEM and the KM271 module to control the Logamatic via MQTT.

Any help or ideas would be highly appreciated.

**Comment by dewenni** [comment]
I do not believe that this is possible. At least I don't know of a suitable command for this so far.
I have also oriented myself in the implementation mainly on the FHEM module and added a few additional commands that I have found elsewhere.
But if you find documentation that describes this at the protocol level, I'm happy to include that, of course.

**Comment by the78mole** [comment]
At least in my ESPhome codebase, there is only a config for the target temperature, not the current temperature.

https://github.com/the78mole/esphome_components/blob/2804c5b55d12be1323ca6402227cdd11328eb567/components/km271_wifi/km271_params.h#L160-L202

But in my opinion, it is not of real interest for the main heating unit. It is a matter of the unit controlling the valve for the radiator in the room. When you have the opening of each valve connected to the heating system, you can adjust the room target temparature and the design temperature (both form the heating curve) accordingly to maximise the openings (water flow) and reduce the flow temperature to a minimum needed to just heat your rooms. This would optimize the systems efficiency and you can use it as a test/preparation to switch e.g., to a heat pump, if you can push the design temperature below 60°C (or better 55°C).

PS: You could maybe tweak it, by changing the [room_temperature_correction](https://github.com/the78mole/esphome_components/blob/2804c5b55d12be1323ca6402227cdd11328eb567/components/km271_wifi/km271_params.h#L167). But I'm not sure, what is the virtual room temperature, when no measurment exists e.g., from the BFU.

**Comment by DanMike** [comment]
Thank you both for your fast responses.

Well, the current room temperature is of interest for the control unit alongside with the outside temperature, to decide whether to burn some oil or not. But of course, I could work around it using other paramters to influence this.

While investigating a bit more, I found this interesting page about the BFU and ist proprietary bus: https://www.holzleitner.com/el/buderus-monitor/index-en.html
This leads to the following conclusions for me:

- There must be a command for setting the current room temperature, as the internal interface for the 2-wire bus to the BFU seems just to mirror everything what's on the system bus.
- But, as long as the BFU ist connected it will overwrite the modified room temperature for sure.
- With a lot of reverse engineering work, a BFU could be simulated on the bus completely, but only if the 2-wire interface is as dumb as assumed. 

Because there's a lot of uncertainty I will go one of the following ways:

1. Try playing around with the other configuration parameters as suggested.
2. Adding a digital potentiometer to the external temperature sensor connector of the BFU to simulate a sensor.The BFU will measure the resistor value, calculate a temperature from it and send this value to the control unit. As most of the digital potentiometers available are controlled via SPI, I could make an extension board for the KM271 WiFi replacement board (just joined the waitlist for the new revision :-) ).

Thanks again for this great projects to you guys!

**Reply by the78mole** [reply]
I have no access to the people on the wait- or wishlist. If you write me a direct message on tindie message, I can send you one of my two reserve boards :-)

**Reply by DanMike** [reply]
Done! :-)

**Reply by ichbinder** [reply]
Hi @DanMike, if you find a solution for the BFU temperature sensor connection I would be very happy if you could publish it. I have the problem that next to the BFU is a fireplace and when I heat this the other half of the house is not heated well or not at all.

**Reply by DanMike** [reply]
@ichbinder : in your case, I would connect an external room temperature sensor (Buderus article no. 5993226) to the BFU and place it elsewhere. After doing some research a software-only solution seems hard to achieve or nearly impossible. My next attempt will be to connect a digital potentiometer to the external sensor input of the BFU. So I could simulate the temperature sensor by controlling the digipot's value via SPI, I2C or OneWire.

**Reply by the78mole** [reply]
If I would have a BFU, I could analyze it. But unfortunately, I have none. Additionally, today I first took a shower with water from my heat pump. I'll soon also switch over the heating circuit to the buffer and connect the oil burner only for cases with extreme low temperatures (or high electricity prices). So, my KM271 will kick in the oil burner (hopefully) only a few times during the winter 😋

**Reply by ichbinder** [reply]
@DanMike thx for this hint, I will get the temperature sensor (Buderus article no. 5993226) and analyze it.

**Reply by DonCanDo** [reply]
Someone know the parameters of the buderus 5993226? May there is another sensor useable
